snapcraft/plugins/go.py

```diff
@@ -101,12 +101,30 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
+        self._go_command = None  # type: str
+        # The build package needs to be installed here. We will only decide
+        # which Go to use during pull however after dependencies have been
+        # staged and build snaps were installed.
         self.build_packages.append('golang-go')
```

Comment:
I was hoping this would take another approach. Similar to what we do in the `python` plugin in `stage_packages` as a property but with `build_packages` here. As in, if we see a `build_snap` entry we do not add `golang-go` to `build_packages` (or by finding it in `stage` using what @kyrofa wrote to find common commands).